app/code/Magento/InventoryInStorePickup/Model/NotifyOrderIsReadyForPickup.php

```diff
@@ -94,27 +105,59 @@ public function __construct(
      *
      * @inheritDoc
      */
-    public function execute(int $orderId): void
+    public function execute(array $orderIds): OperationResultInterface
     {
-        if (!$this->isOrderReadyForPickup->execute($orderId)) {
-            throw new LocalizedException(__('The order is not ready for pickup'));
+        $operationResult = $this->isOrderReadyForPickup->execute($orderIds);
+
+        $success = $operationResult->getSucceeded();
+        $errors = $operationResult->getFailed();
+        foreach ($operationResult->getSucceeded() as $orderId) {
+            try {
+                $order = $this->orderRepository->get($orderId);
+            } catch (NoSuchEntityException $exception) {
+                $errors[] = [
+                    'order_id' => $orderId,
+                    'message' => $exception->getMessage()
+                ];
+                continue;
+            }
+
+            if (!$this->emailNotifier->notify($order)) {
+                $errors[] = [
+                    'order_id' => $orderId,
+                    'message' => __('Cannot notify customer via email.')
+                ];
+                continue;
+            }
+
+            try {
+                $this->shipOrder->execute(
+                    $orderId,
+                    [],
+                    false,
+                    false,
+                    null,
+                    [],
+                    [],
+                    $this->getShipmentArguments($order)
+                );
+                $this->addCommentToOrder->execute($order);
+            } catch (LocalizedException $exception) {
```

Comment:
What if different kind of exception occurs? It might not be possible to recover the orders which were processed or not.